### ISSUE TITLE
Decoding interfaces seems to have broken as of 982b5d9d9190f2b5a2a839137795d831c252adf1

### ISSUE THREAD

**Issue by oxtoacart** [vmihailenco/msgpack#153]
https://github.com/vmihailenco/msgpack/commit/982b5d9d9190f2b5a2a839137795d831c252adf1

[This gist](https://gist.github.com/oxtoacart/7839212896a1e6328fba4fdcdd5555c7) works at revision [358c85424ffc247374779fc5423f4fd4aa882af6](https://github.com/vmihailenco/msgpack/commit/982b5d9d9190f2b5a2a839137795d831c252adf1) but as of revision [982b5d9d9190f2b5a2a839137795d831c252adf1](https://github.com/vmihailenco/msgpack/commit/982b5d9d9190f2b5a2a839137795d831c252adf1) it no longer works and instead fails with:

```
panic: reflect.Set: value of type main.impl is not assignable to type main.Thing

goroutine 1 [running]:
reflect.Value.assignTo(0x10ccdc0, 0x1191bf0, 0x99, 0x10e9653, 0xb, 0x10ce420, 0xc42000e290, 0x0, 0x10ccdc0, 0x1191bf0)
	/usr/local/go/src/reflect/value.go:2186 +0x3a6
reflect.Value.Set(0x10ce420, 0xc42000e290, 0x194, 0x10ccdc0, 0x1191bf0, 0x99)
	/usr/local/go/src/reflect/value.go:1351 +0xa4
gopkg.in/vmihailenco/msgpack%2ev2.(*Decoder).interfaceValue(0xc420086190, 0x10ce420, 0xc42000e290, 0x194, 0x1165300, 0x10ce420)
	/Volumes/external/gocode/src/gopkg.in/vmihailenco/msgpack.v2/decode.go:240 +0x113
gopkg.in/vmihailenco/msgpack%2ev2.decodeInterfaceValue(0xc420086190, 0x10ce420, 0xc42000e290, 0x194, 0x10ce420, 0xc42000e290)
	/Volumes/external/gocode/src/gopkg.in/vmihailenco/msgpack.v2/decode_value.go:193 +0x78
gopkg.in/vmihailenco/msgpack%2ev2.(*Decoder).DecodeValue(0xc420086190, 0x10ce420, 0xc42000e290, 0x194, 0xc42000e290, 0x194)
	/Volumes/external/gocode/src/gopkg.in/vmihailenco/msgpack.v2/decode.go:195 +0x8c
gopkg.in/vmihailenco/msgpack%2ev2.(*Decoder).decode(0xc420086190, 0x10bf020, 0xc42000e290, 0x50, 0x10d9ca0)
	/Volumes/external/gocode/src/gopkg.in/vmihailenco/msgpack.v2/decode.go:190 +0x120
gopkg.in/vmihailenco/msgpack%2ev2.(*Decoder).Decode(0xc420086190, 0xc420047f50, 0x1, 0x1, 0x0, 0x40)
	/Volumes/external/gocode/src/gopkg.in/vmihailenco/msgpack.v2/decode.go:72 +0x6f
gopkg.in/vmihailenco/msgpack%2ev2.Unmarshal(0xc42008c028, 0x3, 0x40, 0xc420047f50, 0x1, 0x1, 0x0, 0x0)
	/Volumes/external/gocode/src/gopkg.in/vmihailenco/msgpack.v2/decode.go:43 +0x1e5
main.main()
	/Volumes/external/gocode/src/msgpack_bug.go:28 +0x1be
```

**Comment by oxtoacart** [comment]
Basically, the value used to unmarshal as a `&thing` and now it's unmarshaling as a `thing`.

**Comment by vmihailenco** [comment]
Should be fixed in v3.2.0

**Comment by oxtoacart** [comment]
Thanks @vmihailenco!
